Code does not compile on rustc 1.74.0

On the latest code (https://github.com/aantn/smag/commit/25091d33606b351bbe0a478c8afd08dd2200361e at time of writing), smag builds successfully on cargo/rustc 1.72.0, but fails on cargo/rustc 1.74.0

Working:
```
$ cargo --version
cargo 1.72.0
$ cargo build
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
```

Broken:
```
$ cargo --version
cargo 1.74.0
$ cargo build
   Compiling smag v0.6.0 (/home/jeremy/src/github.com/aantn/smag)
error[E0283]: type annotations needed
  --> src/ui.rs:19:18
   |
19 |                 .constraints(
   |                  ^^^^^^^^^^^ cannot infer type of the type parameter `C` declared on the method `constraints`
...
24 |                         .as_ref(),
   |                          ------ type must be known at this point
   |
   = note: multiple `impl`s satisfying `Vec<Constraint>: AsRef<_>` found in the `alloc` crate:
           - impl<T, A> AsRef<Vec<T, A>> for Vec<T, A>
             where A: Allocator;
           - impl<T, A> AsRef<[T]> for Vec<T, A>
             where A: Allocator;
help: consider specifying the generic argument
   |
19 |                 .constraints::<&T>(
   |                             ++++++

For more information about this error, try `rustc --explain E0283`.
error: could not compile `smag` (bin "smag") due to previous error
```

--- a/src/ui.rs
+++ b/src/ui.rs
@@ -13,15 +13,14 @@ pub fn draw_ui<T: tui::backend::Backend>(
 ) {
     terminal
         .draw(|f| {
-            let chunks = Layout::default()
+            let chunks: Vec<tui::layout::Rect> = Layout::default()
                 .direction(Direction::Vertical)
                 .margin(2)
                 .constraints(
                     iter::repeat(Constraint::Length(1))
                         .take(args.cmds.len())
                         .chain(iter::once(Constraint::Percentage(10)))
-                        .collect::<Vec<_>>()
-                        .as_ref(),
+                        .collect::<Vec<Constraint>>(),
                 )
                 .split(f.size());
             for (((cmd_id, cmd), stats), &style) in args

